crates/pop-cli/src/commands/up/contract.rs

```diff
@@ -140,4 +147,41 @@ impl UpContractCommand {
 		}
 		Ok(())
 	}
+
+	/// Uploads only the contract without instantiating it.
+	async fn upload_contract(self) -> anyhow::Result<()> {
+		let upload_exec = set_up_upload(UpOpts {
+			path: self.path.clone(),
+			constructor: self.constructor.clone(),
+			args: self.args.clone(),
+			value: self.value.clone(),
+			gas_limit: self.gas_limit,
+			proof_size: self.proof_size,
+			salt: self.salt.clone(),
+			url: self.url.clone(),
+			suri: self.suri.clone(),
+		})
+		.await?;
+		if self.dry_run {
+			match dry_run_upload(&upload_exec).await {
+				Ok(upload_result) => {
+					log::info(format!("Result: {:?}", upload_result.result))?;
```

Comment:
I assume this is creating three log items with a break in between, when it should be a single log entry with line breaks -> e.g. format!("Result: {}\nCode Hash: {}\nDeposit: {}", upload_result.result, upload_result.code_hash, upload_result.deposit)

Reply:
Improved UI:
```
◆  Dry run successful!
│  ● Code Hash: "0xb15348075722f8ac92352b8fcfd6fa3506e2a3f430adadcc79fa73cf23bfe9e7"
│  ● Deposit: "252070000000"
```